l1-contracts/contracts/bridge/L1ERC20Bridge.sol

```diff
@@ -60,73 +77,256 @@ contract L1ERC20Bridge is IL1Bridge, IL1BridgeLegacy, ReentrancyGuard {
 
+    /// @dev The hash of the factory deps, stored here so that the factory deps that each chain provides can be checked when it start the l2 bridge deployment
+    bytes32 internal factoryDepsHash;
+
+    /// @dev A mapping chainId => bridgeProxy. Used to store the bridge proxy's address, and to see if it has been deployed yet.
+    mapping(uint256 => address) public l2BridgeAddress;
+
+    /// @dev A mapping chainId => l2TokenBeacon. Used to store the token beacon proxy's address, and to see if it has been deployed yet.
+    mapping(uint256 => address) public l2TokenBeaconAddress;
+
+    /// @dev A mapping chainId => bridgeProxy. Used to store the bridge proxy's address after deployment has started, but before it has finished.
+    mapping(uint256 => address) internal l2BridgePotentialAddress;
+
+    /// @dev A mapping chainId => l2TokenBeacon. Used to store the token beacon proxy's address after deployment has started, but before it has finished.
+    mapping(uint256 => address) internal l2TokenBeaconPotentialAddress;
+
+    /// @dev we have to record if the bridgeImplTx succeeded
+    mapping(uint256 => bool) internal bridgeImplTxSucceeded;
+
+    /// @dev A mapping chainId => bridgeImplTxHash. Used to check the deploy transaction of the l2Bridge Implementation (which depends on its place in the priority queue).
+    mapping(uint256 => bytes32) internal bridgeImplDeployOnL2TxHash;
+
+    /// @dev A mapping chainId => bridgeProxyTxHash. Used to check the deploy transaction of the l2Bridge Proxy (which depends on its place in the priority queue).
+    mapping(uint256 => bytes32) internal bridgeProxyDeployOnL2TxHash;
+
+    /// @dev A mapping L2 _chainId => Batch number => message number => flag
+    /// @dev Used to indicate that L2 -> L1 message was already processed
+    mapping(uint256 => mapping(uint256 => mapping(uint256 => bool))) public isWithdrawalFinalizedShared;
+
+    /// @dev A mapping chainId => keccak256(account, tokenAddress, amount) => L2 deposit transaction hash => true
+    /// @dev Used for saving the number of deposited funds, to claim them in case the deposit transaction will fail
+    mapping(uint256 => mapping(bytes32 => mapping(bytes32 => bool))) internal depositHappened;
+
+    /// @dev used for extra security until hyperbridging is implemented.
+    mapping(uint256 => mapping(address => uint256)) internal chainBalance;
+
+    /// @dev have we enabled hyperbridging for a given chain yet
+    mapping(uint256 => bool) internal hyperbridgingEnabled;
+
+    /// @dev legacy function gives the l2Bridge address on Era
+    function l2Bridge() external view override returns (address) {
+        return l2BridgeAddress[ERA_CHAIN_ID];
+    }
+
+    /// @dev legacy getter function gives the l2TokenBeacon address on Era
+    function l2TokenBeacon() external view override returns (address) {
+        return ERA_TOKEN_BEACON_ADDRESS;
```

Comment:
Why here is a constant and `l2Bridge` returns `l2BridgeAddress[ERA_CHAIN_ID]`?

Should this return ` l2TokenBeaconAddress[ERA_CHAIN_ID]`?

Reply:
true, wasn't thinking it seems